collect/collect_test.go

```diff
@@ -156,41 +162,49 @@ func TestOriginalSampleRateIsNotedInMetaField(t *testing.T) {
 			Event: types.Event{
 				Dataset:    "aoeu",
 				APIKey:     legacyAPIKey,
-				SampleRate: 50,
+				SampleRate: originalSampleRate,
 				Data:       make(map[string]interface{}),
 			},
 		}
 		coll.AddSpan(span)
 		time.Sleep(conf.SendTickerVal * 5)
 	}
 
+	// At least one event should have been transmitted by now with an original sample rate from upstream sampling.
 	transmission.Mux.RLock()
 	assert.Greater(t, len(transmission.Events), 0, "should be at least one event transmitted")
-	assert.Equal(t, uint(50), transmission.Events[0].Data["meta.refinery.original_sample_rate"],
-		"metadata should be populated with original sample rate")
+	upstreamSampledEvent := transmission.Events[0]
 	transmission.Mux.RUnlock()
 
-	span := &types.Span{
+	assert.Equal(t, originalSampleRate, upstreamSampledEvent.Data["meta.refinery.original_sample_rate"],
+		"metadata should be populated with original sample rate")
+	assert.Equal(t, originalSampleRate*uint(expectedDeterministicSampleRate), upstreamSampledEvent.SampleRate,
+		"sample rate for the event should be the original sample rate multiplied by the deterministic sample rate")
+
+	// Generate one more event with no upstream sampling applied.
+	err = coll.AddSpan(&types.Span{
 		TraceID: fmt.Sprintf("trace-%v", 1000),
 		Event: types.Event{
-			Dataset:    "aoeu",
+			Dataset:    "no-upstream-sampling",
 			APIKey:     legacyAPIKey,
-			SampleRate: 0,
+			SampleRate: 0, // no upstream sampling
 			Data:       make(map[string]interface{}),
 		},
-	}
+	})
+	require.NoError(t, err, "should be able to add the span")
 
-	coll.AddSpan(span)
+	// Find the Refinery-sampled-and-sent event that had no upstream sampling which
+	// should be the last event on the transmission queue.
+	var noUpstreamSampleRateEvent *types.Event
 	assert.Eventually(t, func() bool {
```

Comment:
I know you didn't change this part of the test in this PR. I'm just wondering if using `require` instead of `assert` here would be easier to identify failures here. If the assertion for `"no-upstream-stampling` has failed, the following check won't succeed either.

Reply:
I'm a big fan of `require`—this PR adds it to this test file—and I'm not sure why I didn't use it here.

![nodding kermit](https://media.tenor.com/ncHwlj4_05gAAAAM/nod-kermit-nod.gif)